### ISSUE TITLE
bind to 127.0.0.1

### ISSUE THREAD

**Issue by majimboo** [mscdex/cap#14]
I cannot bind to the loopback adaptor. Is there anything I need to do to be able to?

Saw on your examples that when you did

```
 console.dir(Cap.deviceList());
```

It returned 

```
 //   { name: 'lo',
 //     addresses:
 //      [ { addr: '127.0.0.1', netmask: '255.0.0.0' },
 //        { addr: '::1',
 //          netmask: 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff' } ],
 //     flags: 'PCAP_IF_LOOPBACK' } ]
```

I don't think mine does, its on windows XP. I installed the winpcap libs.

**Comment by freeall** [comment]
@mscdex If I run the first example from the readme on osx and change ip to `127.0.0.1` I get the same issue as mentioned here.

```
$ sudo node foo.js
/private/tmp/foo.js:11
var linkType = c.open(device, filter, bufSize, buffer);
                 ^

TypeError: device must be a string
    at TypeError (native)
    at Object.<anonymous> (/private/tmp/foo.js:11:18)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:475:10)
    at startup (node.js:117:18)
    at node.js:951:3
```

The code is the same as the one in the readme, just added here for clarity:

```
var Cap = require('cap').Cap,
    decoders = require('cap').decoders,
    PROTOCOL = decoders.PROTOCOL;

var c = new Cap(),
    device = Cap.findDevice('127.0.0.1'),
    filter = 'tcp and dst port 80',
    bufSize = 10 * 1024 * 1024,
    buffer = new Buffer(65535);

var linkType = c.open(device, filter, bufSize, buffer);

c.setMinBytes && c.setMinBytes(0);

c.on('packet', function(nbytes, trunc) {
  console.log('packet: length ' + nbytes + ' bytes, truncated? '
              + (trunc ? 'yes' : 'no'));

  // raw packet data === buffer.slice(0, nbytes)

  if (linkType === 'ETHERNET') {
    var ret = decoders.Ethernet(buffer);

    if (ret.info.type === PROTOCOL.ETHERNET.IPV4) {
      console.log('Decoding IPv4 ...');

      ret = decoders.IPV4(buffer, ret.offset);
      console.log('from: ' + ret.info.srcaddr + ' to ' + ret.info.dstaddr);

      if (ret.info.protocol === PROTOCOL.IP.TCP) {
        var datalen = ret.info.totallen - ret.hdrlen;

        console.log('Decoding TCP ...');

        ret = decoders.TCP(buffer, ret.offset);
        console.log(' from port: ' + ret.info.srcport + ' to port: ' + ret.info.dstport);
        datalen -= ret.hdrlen;
        console.log(buffer.toString('binary', ret.offset, ret.offset + datalen));
      } else if (ret.info.protocol === PROTOCOL.IP.UDP) {
        console.log('Decoding UDP ...');

        ret = decoders.UDP(buffer, ret.offset);
        console.log(' from port: ' + ret.info.srcport + ' to port: ' + ret.info.dstport);
        console.log(buffer.toString('binary', ret.offset, ret.offset + ret.info.length));
      } else
        console.log('Unsupported IPv4 protocol: ' + PROTOCOL.IP[ret.info.protocol]);
    } else
      console.log('Unsupported Ethertype: ' + PROTOCOL.ETHERNET[ret.info.type]);
  }
});
```

**Comment by mscdex** [comment]
@freeall This should be fixed in 544c65c17 but you will need to also make sure you run with `sudo` to make sure you can get available network interface information (and start actual packet capturing). Although I know on linux you can avoid `sudo` by using `setcap`, but it needs to be done on the node executable IIRC which may not be desirable. On OS X I think you can just change the permissions on the `/dev/bpf*` devices?

**Comment by freeall** [comment]
@mscdex Thanks a lot for fixing really fast! Now we can continue working on this module we're doing where we want to read all streams. Will probably come out later tonight if it works well.
